File: typescript/easy/staircase.ts
function fillSpaces(fill: number, value: string): string {
  let space = "";
  for (let i = 1; i <= fill; i++) {
      space += value;
  }
  return space;
}

function staircase(n: number): void {
  for (let i = 1; i <= n; i++) {
      const whiteSpaces = fillSpaces(n - i, " ");
      const hashSpaces = fillSpaces(i, "#");
      console.log(whiteSpaces + hashSpaces);
  }
}

staircase(6);

File: easy/golang/staircase.go
package main

import (
	"fmt"
	"strings"
)

func fillSpaces(fill int, value string) string {
	return strings.Repeat(value, fill)
}

func staircase(n int) {
	for i := 1; i <= n; i++ {
		whiteSpaces := fillSpaces(n-i, " ")
		hashSpaces := fillSpaces(i, "#")
		fmt.Println(whiteSpaces + hashSpaces)
	}
}

func main() {
	staircase(6)
}
